lib/sort-order-selector/index.tsx

```diff
@@ -0,0 +1,122 @@
+import React, { Fragment, FunctionComponent, useEffect, useRef } from 'react';
+import { connect } from 'react-redux';
+import { toggleSortOrder, setSortType } from '../state/settings/actions';
+import { recordEvent } from '../state/analytics/middleware';
+
+import * as S from '../state';
+import * as T from '../types';
+
+type StateProps = {
+  shouldDisplay: boolean;
+  sortReversed: boolean;
+  sortType: T.SortType;
+};
+
+type DispatchProps = {
+  setSortType: (sortType: T.SortType) => any;
+  toggleSortOrder: () => any;
+};
+
+type Props = StateProps & DispatchProps;
+
+export const SortOrderSelector: FunctionComponent<Props> = ({
+  shouldDisplay,
+  setSortType,
+  sortReversed,
+  sortType,
+  toggleSortOrder,
+}) => {
+  const changeSort = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    const selectedOption = event.currentTarget.selectedOptions[0];
+    const selectedSortType = selectedOption.dataset.order;
+    const selectedSortReversed = selectedOption.dataset.reversed;
+
+    if (selectedSortType !== sortType) {
+      setSortType(selectedSortType as T.SortType);
+    }
+    if ((selectedSortReversed === 'false') === sortReversed) {
```

Comment:
this is confusing me, specifically because we're comparing against the string value of `false`

I think we're doing it funny-like because it's a string value and not a real value. in this case _parsing_ it could go a long way for readability. what I expect is to find something like `selectedSortReversed !== sortReversed`

```ts
const selectedSortReversed = selectedOption.dataset.reversed === 'true'; // true if 'true' else false
const selectedSortReversed = JSON.parse(selectedOption.dataset.reversed); // 'true' and 'false' decode properly

if (selectedSortReversed !== sortReversed) {
	…
}
```

Reply:
Oops, I should have refactored that out actually.